src/plonkish/backend/halo2.rs

```diff
@@ -388,13 +415,35 @@ pub struct ChiquitoHalo2SuperCircuit<F: Field + From<u64>> {
     witness: SuperAssignments<F>,
 }
 
-impl<F: Field + From<u64>> ChiquitoHalo2SuperCircuit<F> {
+impl<F: Field + From<u64> + Hash> ChiquitoHalo2SuperCircuit<F> {
     pub fn new(sub_circuits: Vec<ChiquitoHalo2<F>>, witness: SuperAssignments<F>) -> Self {
         Self {
             sub_circuits,
             witness,
         }
     }
+
+    pub fn instance(&self) -> Vec<Vec<F>> {
+        if !self.witness.is_empty() {
+            let mut result = Vec::new();
+
+            for (uuid, assignment) in &self.witness {
```

Comment:
@qwang98 I don't think you should follow the order of the witness, but the order of sub_circuit, or whatever order is used to call meta.instance_column()

Reply:
Agreed. sub_circuit is the order meta.instance_column is called.